trie-db/src/proof/generate.rs

```diff
@@ -75,13 +88,34 @@ impl<'a, C: NodeCodec> StackEntry<'a, C> {
 
 	/// Encode this entry to an encoded trie node with data properly omitted.
 	fn encode_node(mut self) -> TrieResult<Vec<u8>, C::HashOut, C::Error> {
+		let omit_value = self.omit_value;
+		let prepend_on_hashed = |mut encoded: Vec<u8>| -> Vec<u8> {
+			if let Some(header) = C::ESCAPE_HEADER {
+				if matches!(omit_value, ValueMgmt::OmitHashedValue) {
+					for i in header.iter() {
+						encoded.insert(0, *i);
+					}
+				}
+			}
+			encoded
+		};
 		let node_data = self.node.data();
-		Ok(match self.node.node_plan() {
+		let value_with_omission = |
+			value_range: &ValuePlan,
+		| -> Value
+		{
+			if !matches!(omit_value, ValueMgmt::Standard) {
+				Value::NoValue
+			} else {
+				value_range.build(&node_data)
+			}
+		};
+		let encoded = match self.node.node_plan() {
 			NodePlan::Empty => node_data.to_vec(),
-			NodePlan::Leaf { .. } if !self.omit_value => node_data.to_vec(),
+			NodePlan::Leaf { .. } if matches!(omit_value, ValueMgmt::Standard) => node_data.to_vec(),
 			NodePlan::Leaf { partial, value: _ } => {
 				let partial = partial.build(node_data);
-				C::leaf_node(partial.right(), &[])
+				prepend_on_hashed(C::leaf_node(partial.right(), Value::Value(&[])))
```

Comment:
can a LeafNode contains a HashedValue ?

Reply:
Yes, here we remove the value because it is a skippend one (in input when verifying), the matching above should be use when we do not skip it.